test/promql/tests/data_plane/gitops_cluster_agent_operation_test.yaml

```diff
@@ -0,0 +1,59 @@
+evaluation_interval: 1m
+
+rule_files:
+  - prometheus.gitops_cluster_agent_op_alerts.yaml
+
+tests:
+  # ----- Cluster-Agent operations in non-complete state -----
+  - interval: 1m
+    input_series:
+      # More than 10 cluster-agent operations are in non-complete state, should alert
+      - series: 'operationDB_rows_in_non_complete_state{source_cluster="cluster01"}'
+        values: '11x11'   # 11 cluster-agent operations are in non-complete state
+
+    alert_rule_test:
+      - eval_time: 15m
+        alertname: GitOpsClusterAgentOperationErrors
+        exp_alerts:
+          - exp_labels:
+              severity: warning
+              slo: true
+              source_cluster: cluster01
+            exp_annotations:
+              summary: >-
+                Cluster-Agent Operations in non-complete state are too high. Got: 11
+              description: >-
+                The sum of cluster-agent operations in non-complete state is greater than 10 on cluster cluster01
+              runbook_url: https://gitlab.cee.redhat.com/rhtap/docs/sop/-/blob/main/gitops/cluster-agent-operations.md
+
+  - interval: 1m
+    input_series:
+      # 10 cluster-agent operations are in non-complete state, should alert
+      - series: 'operationDB_rows_in_non_complete_state{source_cluster="cluster01"}'
+        values: '10x10'   # 10 cluster-agent operations are in non-complete state
+
+    alert_rule_test:
+      - eval_time: 15m
+        alertname: GitOpsClusterAgentOperationErrors
+        exp_alerts:
+          - exp_labels:
+              severity: warning
+              slo: true
+              source_cluster: cluster01
+            exp_annotations:
+              summary: >-
+                Cluster-Agent Operations in non-complete state are too high. Got: 10
+              description: >-
+                The sum of cluster-agent operations in non-complete state is greater than 10 on cluster cluster01
+              runbook_url: https://gitlab.cee.redhat.com/rhtap/docs/sop/-/blob/main/gitops/cluster-agent-operations.md
+
+  - interval: 1m
+    input_series:
+
+      # Less than 10 cluster-agent operations are in non-complete state, should not alert
+      - series: 'operationDB_rows_in_non_complete_state{source_cluster="cluster01"}'
+        values: '5x5'   # 5 cluster-agent operations are in non-complete state
+
+    alert_rule_test:
+      - eval_time: 15m
```

Comment:
The series is:   5 5 5 5 5
but the eval_time is 15m, although this test does pass, is this the intention of the test? @yftacherzog, should we provide a time series that exceeds the `eval_time` and the 'for' duration? (Just looking at your original review comment for this specific test)

Reply:
Yes. We should have the time series at least as long as the eval_time. If we want to test missing-data scenarios, then we should have `_` "measurements" instead of numbers for where the data is missing.